src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/AbstractDefaultGetCaseViewOperation.java

```diff
@@ -85,4 +102,18 @@ CaseTabCollection getCaseTabCollection(String caseTypeId) {
             .collect(Collectors.toList());
     }
 
+    void hydrateHistoryField(CaseDetails caseDetails, CaseType caseType, List<CaseViewEvent> events) {
+        final List<CaseField> caseFieldDefinitions = caseType.getCaseFields();
+        final JsonNode data = MAPPER.convertValue(caseDetails.getData(), JsonNode.class);
+        if (data != null) {
```

Comment:
What is the use of data variable? It is not clear.

Reply:
This is to build a value for a field of type CaseHistoryViewer. This is making the case history tab a flexible control.